flux-refineck/src/checker.rs

```diff
@@ -972,7 +972,8 @@ impl<'a, 'tcx, M: Mode> Checker<'a, 'tcx, M> {
             }
             CastKind::FloatToInt
             | CastKind::IntToFloat
-            | CastKind::Pointer(mir::PointerCast::MutToConstPointer) => {
+            | CastKind::Pointer(mir::PointerCast::MutToConstPointer)
+            | CastKind::Pointer(mir::PointerCast::Unsize) => {
```

Comment:
This is unsound for `mir::PointerCast::Unsize`. As implemented, you can use it to coerce a `&mut [i32{v: v > 0}; 10]` to a `&mut [i32]`. I tried to implement this once but got caught up in the many details because there are other forms of [unsized coercions ](https://doc.rust-lang.org/reference/type-coercions.html#unsized-coercions). 

We should make sure this is only being applied for unsizing of arrays into slices (I think this is enough for panic) and that the semantics is

```
&mut [T; n] -> &mut [T][n]
&[T;n] -> &[T][n]
```

Related to https://github.com/flux-rs/flux/issues/285

Reply:
ok I made some changes -- see if its what you had in mind?